test/com/google/enterprise/adaptor/prebuilt/DateFilterTest.java

```diff
@@ -0,0 +1,465 @@
+    assertEquals(null, params.get("Transmission-Decision"));
+  }
+
+  @Test
+  public void testTransform_DateInMetadata() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("days", "365");
+    config.put("keyset", "metadata");
+    DateFilter transform = DateFilter.create(config);
+    Map<String, String> params = new HashMap<String, String>();
+    Metadata metadata = new Metadata();
+    metadata.add("lastModified", "1977-06-18");
+    transform.transform(metadata, params);
+    assertEquals("do-not-index", params.get("Transmission-Decision"));
+  }
+
+  @Test
+  public void testTransform_RecentDateInMetadata() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("days", "365");
+    config.put("keyset", "metadata");
+    DateFilter transform = DateFilter.create(config);
+    Map<String, String> params = new HashMap<String, String>();
+    Metadata metadata = new Metadata();
+    metadata.add("lastModified", dateFormat.format(new Date()));
+    transform.transform(metadata, params);
+    assertEquals(null, params.get("Transmission-Decision"));
+  }
+
+  @Test
+  public void testTransform_MultipleDatesInMetadata() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "modifiedHistory");
+    config.put("days", "365");
+    config.put("keyset", "metadata");
+    DateFilter transform = DateFilter.create(config);
+    Map<String, String> params = new HashMap<String, String>();
+    Metadata metadata = new Metadata();
+    metadata.add("modifiedHistory", dateFormat.format(new Date()));
+    metadata.add("modifiedHistory", "1977-06-18");
+    transform.transform(metadata, params);
+    assertEquals("do-not-index", params.get("Transmission-Decision"));
+  }
+
+  @Test
+  public void testTransform_MultipleDatesInMetadataOneBad() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "modifiedHistory");
+    config.put("days", "365");
+    config.put("keyset", "metadata");
+    DateFilter transform = DateFilter.create(config);
+    Map<String, String> params = new HashMap<String, String>();
+    Metadata metadata = new Metadata();
+    metadata.add("modifiedHistory", "07/21/1969");
+    metadata.add("modifiedHistory", "1977-06-18");
+    transform.transform(metadata, params);
+    assertEquals("do-not-index", params.get("Transmission-Decision"));
+  }
+
+  @Test
+  public void testTransform_DateInMetadataAndParams() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("days", "365");
+    DateFilter transform = DateFilter.create(config);
+    Map<String, String> params = new HashMap<String, String>();
+    params.put("lastModified", dateFormat.format(new Date()));
+    Metadata metadata = new Metadata();
+    metadata.add("lastModified", "1977-06-18");
```

Comment:
Maybe flipping the dates would be better, because if "metadata or params" were still in play, it would always skip this document, but now it would not be skipped if the old date were in params.